### PR TITLE
Added typescript native delegate example

### PR DESCRIPTION
**Pull request by roblav96**
Took me a while to fully understand how to do this in typescript.

### CODE REVIEW

File: how-to/ObjC-Subclassing.md [hunk 1]
```diff
@@ -182,8 +182,68 @@ class JSObject extends NSObject implements NSCoding {
 There should be no TypeScript constructor, because it will not be executed. Instead override one of the `init` methods.
 
 
+## TypeScript Delegate Example
+
+When working with native APIs, you'll find yourself having to setup delegates in order to recieve results or callbacks. For this example, we'll setup a delegate for the [Tesseract-OCR-iOS](https://github.com/gali8/Tesseract-OCR-iOS/wiki/Using-Tesseract-OCR-iOS#using-tesseract-object) API.
```
Comment (ivanbuhov) [inline how-to/ObjC-Subclassing.md]: Just to make sure that if one day the [Tesseract OCR API](https://github.com/gali8/Tesseract-OCR-iOS/wiki/Using-Tesseract-OCR-iOS) is changed we are still linking the old API version... what about linking the [exact revision](https://github.com/gali8/Tesseract-OCR-iOS/wiki/Using-Tesseract-OCR-iOS/6510b29bbf18655f29a26f484b00a24cc66ed88b) of the wiki.

File: how-to/ObjC-Subclassing.md [hunk 2]
```diff
@@ -182,8 +182,68 @@ class JSObject extends NSObject implements NSCoding {
...
+
+}
+```
+
+Now that we have our delegate class setup, we can create a new instance of `G8Tesseract` and start using it:
+```typescript
+function image2text(image: UIImage): string {
+	let delegate: G8TesseractDelegateImpl = G8TesseractDelegateImpl.new()
+	let tess: G8Tesseract = G8Tesseract.new()
+	tess.delegate = delegate
```
Comment (ivanbuhov) [inline how-to/ObjC-Subclassing.md]: We can add a comment above the line to emphasize on the fact that the `delegate` property is weak. Maybe something like:
_"This property is weak and won't be retained by the Objective-C runtime so you should manually keep the delegate JS object alive as long the tessaract instance is alive"_
It is not what the example is trying to demonstrated but IMO it is worth mentioning.

Thanks for the contribution :)

**Comment by roblav96** [comment]
@ivanbuhov Thanks for the pointers! That's an excellent point you've made about

> "This property is weak and won't be retained by the Objective-C runtime so you should manually keep the delegate JS object alive as long the tessaract instance is alive"

I had no idea about that. Thank you! Are there any modifications you need me to make?

**Comment by ivanbuhov** [comment]
@roblav96 I find these two modifications to be enough:
1. Changing the [link to the Tesseract wiki](https://github.com/NativeScript/ios-runtime-docs/pull/56/files#diff-cfe8c22f055fe5565123a1921fbdad95R187) from `https://github.com/gali8/Tesseract-OCR-iOS/wiki/Using-Tesseract-OCR-iOS#using-tesseract-object` to `https://github.com/gali8/Tesseract-OCR-iOS/wiki/Using-Tesseract-OCR-iOS/6510b29bbf18655f29a26f484b00a24cc66ed88b`
2. Add [this comment](https://github.com/NativeScript/ios-runtime-docs/pull/56#discussion_r78356498) as JavaScript comment in the example above [this line](https://github.com/NativeScript/ios-runtime-docs/pull/56/files#diff-cfe8c22f055fe5565123a1921fbdad95R232).

Thanks again for your contribution :)

**Comment by roblav96** [comment]
@ivanbuhov @jasssonpet Hey guys, totally forgot all about this one ☁️

Let me know if there are any other revisions needed. Thanks!

**Pull request merged**